### PR TITLE
Add missing VOF term in the strong form of the residual

### PR DESCRIPTION
**Pull request by blaisb**
# Description of the problem

- There was a missing term in the strong form of the residual which is (mu_1 - mu_0) * (grad_u + grad_u^T) * grad_phase
- This does not have significant consequences, but I think it is best to remain consistent

# Description of the solution

- Add missing term

# How Has This Been Tested?

- All test results barely change, but I had to update them nonetheless
- It seems to help a little bit in terms of velocity divergence, but really not much to be honest.

### CODE REVIEW

File: source/solvers/navier_stokes_vof_assemblers.cc [hunk 1]
```diff
@@ -64,6 +68,8 @@ GLSNavierStokesVOFAssemblerCore<dim>::assemble_matrix(
       const Tensor<1, dim> pressure_gradient =
         scratch_data.pressure_gradients[q];
 
+
+
```
Comment (blaisb) [inline source/solvers/navier_stokes_vof_assemblers.cc:72]:
```suggestion
```

File: source/solvers/navier_stokes_vof_assemblers.cc [hunk 2]
```diff
@@ -143,7 +157,7 @@ GLSNavierStokesVOFAssemblerCore<dim>::assemble_matrix(
               const auto &phi_u_j      = scratch_data.phi_u[q][j];
               const auto &grad_phi_u_j = scratch_data.grad_phi_u[q][j];
               const auto &div_phi_u_j  = scratch_data.div_phi_u[q][j];
-              const auto &shear_rate_j = grad_phi_u_j + transpose(grad_phi_u_j);
+              const auto  shear_rate_j = grad_phi_u_j + transpose(grad_phi_u_j);
```
Comment (blaisb) [inline source/solvers/navier_stokes_vof_assemblers.cc:160]: This should not have been a reference, so this was a bit weird.

File: source/solvers/navier_stokes_vof_assemblers.cc [hunk 3]
```diff
@@ -90,12 +94,22 @@ GLSNavierStokesVOFAssemblerCore<dim>::assemble_matrix(
           calculate_navier_stokes_gls_tau_transient(
             u_mag, viscosity_for_stabilization_vector[q] / density_eq, h, sdt);
 
+      // Void fraction gradient for strong residual
```
Comment (AmishgaAlphonius) [inline source/solvers/navier_stokes_vof_assemblers.cc:97]:
```suggestion
      // Filtered phase fraction gradient for strong residual
```

File: source/solvers/navier_stokes_vof_assemblers.cc [hunk 4]
```diff
@@ -243,6 +256,10 @@ GLSNavierStokesVOFAssemblerCore<dim>::assemble_rhs(
       const Tensor<1, dim> pressure_gradient =
         scratch_data.pressure_gradients[q];
 
+      // Void fraction gradient for strong residual
```
Comment (AmishgaAlphonius) [inline source/solvers/navier_stokes_vof_assemblers.cc:259]:
```suggestion
      // Filtered phase fraction gradient for strong residual
```

File: source/solvers/navier_stokes_vof_assemblers.cc [hunk 5]
```diff
@@ -909,12 +932,22 @@ GLSNavierStokesVOFAssemblerNonNewtonianCore<dim>::assemble_matrix(
           calculate_navier_stokes_gls_tau_transient(
             u_mag, viscosity_for_stabilization_vector[q] / density_eq, h, sdt);
 
+      // Void fraction gradient for strong residual
```
Comment (AmishgaAlphonius) [inline source/solvers/navier_stokes_vof_assemblers.cc:935]:
```suggestion
      // Filtered phase fraction gradient for strong residual
```

File: source/solvers/navier_stokes_vof_assemblers.cc [hunk 6]
```diff
@@ -1107,12 +1140,22 @@ GLSNavierStokesVOFAssemblerNonNewtonianCore<dim>::assemble_rhs(
             u_mag, viscosity_for_stabilization_vector[q] / density_eq, h, sdt);
 
 
+      // Void fraction gradient for strong residual
```
Comment (AmishgaAlphonius) [inline source/solvers/navier_stokes_vof_assemblers.cc:1143]:
```suggestion
      // Filtered phase fraction gradient for strong residual
```

**Review by AmishgaAlphonius — APPROVED** [review]
Funny that it was not thought of before! I only found a small typo to fix :)

File: source/solvers/navier_stokes_vof_assemblers.cc (hunk 3 above)
Comment (lpsaavedra) [inline source/solvers/navier_stokes_vof_assemblers.cc:97]:
```suggestion
      // Filtered phase fraction gradient for strong residual
```

File: source/solvers/navier_stokes_vof_assemblers.cc (hunk 4 above)
Comment (lpsaavedra) [inline source/solvers/navier_stokes_vof_assemblers.cc:259]:
```suggestion
      // Filtered phase fraction gradient for strong residual
```

File: source/solvers/navier_stokes_vof_assemblers.cc (hunk 5 above)
Comment (lpsaavedra) [inline source/solvers/navier_stokes_vof_assemblers.cc:935]:
```suggestion
      // Filtered phase fraction gradient for strong residual
```

File: source/solvers/navier_stokes_vof_assemblers.cc (hunk 6 above)
Comment (lpsaavedra) [inline source/solvers/navier_stokes_vof_assemblers.cc:1143]:
```suggestion
      // Filtered phase fraction gradient for strong residual
```

**Review by lpsaavedra — APPROVED** [review]
I have only a small comment. These equations are not in the documentation, am I correct? I guess we should be writing these equations there soon in a "multiphysics coupling" section...

**Review by PierreLaurentinCS — CHANGES_REQUESTED** [review]
Nothing to say, everything is clean, but I'd like this question answered before merging : 
Why don't we add the term also in the strong jacobian ?

**Review by hepap — APPROVED** [review]

File: applications_tests/lethe-fluid/time_dependent_boundaries_vof.output [hunk 7]
```diff
@@ -24,26 +24,26 @@ Transient iteration: 1        Time: 0.5      Time step: 0.5      CFL: 2.79578
...
+   time    surface_fluid_0 mass_per_length_fluid_0 momentum-x_fluid_0 momentum-y_fluid_0 surface_fluid_1 mass_per_length_fluid_1 momentum-x_fluid_1 momentum-y_fluid_1 sharpening_threshold 
+1.0000e+00      7.9960e-01              7.9960e-02         3.3587e-02         2.9026e-02      2.0040e-01              2.0040e+01         4.9507e+00         1.9483e-01           5.0000e-01 
 
 ---------------
 VOF Barycenter
 ---------------
-  time    x_vof    y_vof    vx_vof   vy_vof
-1.000000 0.513510 0.638807 0.449408 -0.013436
+  time    x_vof    y_vof    vx_vof   vy_vof  
+1.000000 0.512900 0.576332 0.363353 0.013339
```
[the remark this answers is not in the document]
Comment (blaisb) [reply applications_tests/lethe-fluid/time_dependent_boundaries_vof.output:49]: Yeah I was wondering this also. I'm not exactly sure what's up. I'll double check.

**Comment by blaisb** [comment]
> Nothing to say, everything is clean, but I'd like this question answered before merging : Why don't we add the term also in the strong jacobian ?

Good question. The strong Jacobian term is very expensive to calculate since it has the full hessian. I checked many cases and it never really affects convergence, so for now we can neglect it i think.

**Review by PierreLaurentinCS — APPROVED** [review]

File: applications_tests/lethe-fluid/time_dependent_boundaries_vof.output (hunk 7 above)
[the remark this answers is not in the document]
Comment (blaisb) [reply applications_tests/lethe-fluid/time_dependent_boundaries_vof.output:49]: Yeah that test is weird as fuck, I'm not surpirsed the results changed that much, the test has not real physical meaning it's just there to see if the boundary change in time. No alarm to be had.

**Pull request merged**